pages/posts/stability.md

```diff
@@ -0,0 +1,137 @@
+and forward compatibility requirements that must be followed for any change. It
+will be much like JavaScript in that as it evolves you can always be sure that
+your existing schemas will continue to work with any implementation you're
+using, but you take a risk using newer features because not all implementations
+will have implemented those features yet.
+
+That vision of a stable yet continuously evolving spec doesn't fit well with the
+IETF process. There are paths we considered, but nothing was proposed that we
+thought would allow us to continue to evolve the standard and get out of "draft"
+any time soon. Therefore, the first step to achieving our vision is to decouple
+from IETF. That allows us to pursue a new model that is more conducive to our
+vision.
+
+Since we want JSON Schema to evolve similarly to JavaScript, we are modeling our
+new process off of the ECMAScript TC39 process. In our next release, most of the
+keywords and features you use today will be declared stable and they will never
+change in a backward incompatible way again. Features and keywords we aren't
+comfortable making stable yet will become part of a new staged release process
+that we are working on defining now. The goal of the staged release process is
+to ensure that the feature gets sufficient implementation, testing, and real
+world vetting for us to be confident in declaring it stable.
+
+## Standards considerations
+
+Starting with the next release, the JSON Schema specification will be
+self-published on our website. One of the concerns about self-publishing is
+about other standards being able to reference the JSON Schema specification. We
+have received feedback from people involved in standards development that our
+approach would be acceptable for them to reference our specification in their
+standards based on our membership with the OpenJS Foundation. We don't know if
+_all_ standards bodies will come to the same conclusion, but this feedback gives
+us confidence that it isn't going to be a significant problem.
+
+## The bottom line
+
+Details about the new process will be shared in separate post when they are
+finalized, but here are some outcomes users can expect.
+
+* If you use only stable features, you are guaranteed interoperability between
+  implementations and you will never need to update those schemas just for the
+  sake of keeping up with new releases.
+* You can use new features before they are stable, but you should only do so if
+  you are in a closed environment where the implementation(s) you will be using
+  support the feature.
+* Compatibility/Interoperability guarantees will only apply to the next release
+  and beyond. You will need to update your schemas to the stable version, but
+  you won't have to continue to update them as JSON Schema evolves.
+* Custom dialects will still be a thing, but the vocabulary system will
+  initially be considered "unstable" and is likely to change.
```

Comment:
I'd like to discuss the messaging here because this places too much emphasis on the vocabulary selection mechanism and not enough emphasis on the role of vocabularies in new keyword development.

We have a plan of how to handle new keyword proposals, and it relies on vocabularies.  "How do I get my keyword proposal considered?" is a major question of interest, and the new approach under the SDLC will offer far better guarantees and guidance on how to do that.  That should be a major point.

Focusing on how `$vocabulary` itself is considered unstable gives the impression that vocabularies are relegated to a weird experimental second-class status, which is not what we want.  How are people supposed to feel confident in the evolution of JSON Schema if the only mention of the system we expect to use to evaluate new keywords is that it will still be "a thing" but "unstable."

Vocabularies as a concept are not unstable.  The `$vocabulary` keyword is.  The placement of `$vocabulary` (or some other control) in the meta-schema can be considered unstable.  The granularity of vocabularies can be considered unstable.

But the existence of custom dialects is not.  Their use cases are not.  Their role in new keyword development and evaluation is not.

Reply:
I think the role of vocabularies in our process is out of scope for this post and I thought I was clear that it was the "vocabulary system" that would be unstable, not "vocabularies" as a concept.

What I'm trying to convey here is that the concept of custom dialects and vocabularies isn't going away, but the mechanism for how you define them is still evolving. Maybe it should be expressed more like that without mentioning the vocabulary system or stability specifically.

Reply:
> What I'm trying to convey here is that the concept of custom dialects and vocabularies isn't going away, but the mechanism for how you define them is still evolving. Maybe it should be expressed more like that without mentioning the vocabulary system or stability specifically.

Yes, in the current version the vocabulary system is the _only_ thing specifically singled out as unstable, and the phrasing around it suggests that its use should be minimized.

> I thought I was clear that it was the "vocabulary system" that would be unstable, not "vocabularies" as a concept.

I don't think that distinction is very clear to most people.

Reply:
Given you've sinced approved this PR, I'm marking this thread resolved.